src/plugins/customer-order/generate-order/generate-order.resolver.ts

```diff
@@ -0,0 +1,112 @@
+                    // se verifica que se pueda asignar el producto, de lo contrario omitimos agregar el producto a la orden
+                    if (jsonValue && (productVariant.stockOnHand - productVariant.stockAllocated) > 0) {
+                        try{
+                            let addItemResult: ErrorResultUnion<Order, any> = await this.orderService.addItemToOrder(ctx, order?.id as ID, productVariant.id, jsonValue.quantity);
+                            if (addItemResult instanceof Order){
+                                order = addItemResult;
+                            } else if(addItemResult.order){;
+                                order = addItemResult.order
+                            }
+                        }catch(e){/** error to add products to order beacuse insuficient stock  */}
+                    }
+                };
+            }catch(e) {console.log(`ERROR ADDING ITEMS TO ORDER, detail: ${e}`);}
+
+            if (order) {
+                if (order.lines.length > 0) {
+                    try {
+                        let transitionResult = await this.orderService.transitionToState(ctx, order.id, 'ArrangingPayment');
+                    } catch (e) {
+                        try{
+                            await this.orderService.cancelOrder(ctx,{orderId: order.id,reason: "ERROR on transi¡tion state to ArrangingPayment" });
+                        } catch(e){}
+                        throw new IllegalOperationError("Ocurrio un error al procesar el pedido");
+                    }
+
+                    try {
+                        let paymentMethods = await this.orderService.getEligiblePaymentMethods(ctx, order.id);
+                        if (paymentMethods.length > 0 && paymentMethods[0].isEligible) {
+                            order = await this.orderService.addPaymentToOrder(
+                                ctx,
+                                order.id,
+                                { method: paymentMethods[0].code, metadata: { data: "PAYMENT METHOD FOR DJ-BOT" } });
+                        } else {
+                            order = await this.orderService.cancelOrder(ctx,{orderId: order.id,reason: "CANCEL ORDER, NO PAYMENT TYPE AVAILABLE"});
+                        }
+                    } catch (e) {console.log("Error on transition order state or set paymentMethod");}
+                    if (order.state != 'PaymentAuthorized' || order.lines.length == 0) {
+                        throw new IllegalOperationError("Ocurrio un error al autorizar el pedido");
+                    }
+                }
+            }
+
+        } else {
+            if (search.length == 0){
+                throw new IllegalOperationError("No encontramos los productos solicitados");
+            } else {
+                throw new IllegalOperationError("Stock insuficiente para generar pedido");
+            }
```

Comment:
No tenemos un catch para estos tal vez logguear el mensaje podría ayudar a saber cuantos pedidos tienen problemas, independientemente del objeto que regrese la GraphQL...
Por otro lado podríamos refactorizar tal vez esto así:
```suggestion
          throw new IllegalOperationError(search.length == 0 ? 'No encontramos los productos solicitados' : 'Stock insuficiente para generar pedido');
```